crates/matrix-sdk/src/oidc/registrations.rs

```diff
@@ -20,22 +28,24 @@ use std::{
     path::PathBuf,
 };
 
-use matrix_sdk::oidc::types::registration::{
+use mas_oidc_client::types::registration::{
     ClientMetadata, ClientMetadataVerificationError, VerifiedClientMetadata,
 };
 use serde::{Deserialize, Serialize};
 use url::Url;
 
+/// Errors related to persisting OIDC registrations.
 #[derive(Debug, thiserror::Error)]
 pub enum OidcRegistrationsError {
+    /// The supplied base path is invalid.
     #[error("Failed to use the supplied base path.")]
     InvalidBasePath,
-    #[error("Failed to load registrations file: {message}")]
-    LoadFailure { message: String },
-    #[error("Metadata mismatch, ignoring any stored registrations.")]
-    MetadataMismatch,
+    /// An error occurred whilst saving the registration data.
     #[error("Failed to save the registration data {message}.")]
-    SaveFailure { message: String },
+    SaveFailure {
+        /// The error message.
+        message: String,
+    },
```

Comment:
Could this be instead `SaveFailure(#[source] Box<dyn std::error::Error>)`? So the end user can pattern match the precise error type if they want to, instead of getting only a message (that might be incomplete).

Reply:
Ah yeah I wondered about this when I added the docs but forgot to add a comment. Thanks 🙌